### PR TITLE
Add AutoTest for scala examples

### PR DESCRIPTION
**Pull request by respecteverything**
Add Auto Test for:

- [x] analytics-zoo/apps : model-inference-examples
- [x] analytics-zoo/zoo/src/main/scala/com/intel/analytics/zoo/examples : chatbot
- [x] analytics-zoo/zoo/src/main/scala/com/intel/analytics/zoo/examples : tfnet
- [x] analytics-zoo/zoo/src/main/scala/com/intel/analytics/zoo/examples : Streaming
- [x] analytics-zoo/zoo/src/main/scala/com/intel/analytics/zoo/examples : Local estimator

### CODE REVIEW

File: zoo/src/main/scala/com/intel/analytics/zoo/examples/run-scala-example-test.sh [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+else
+    wget $FTP_URI/analytics-zoo-data/data/ -P analytics-zoo-data/data
+    unzip -q analytics-zoo-data/data/object-detection-coco.zip -d analytics-zoo-data/data/object-detection-coco
+fi
+
+bash ${ANALYTICS_ZOO_HOME}/bin/spark-shell-with-zoo.sh \
+--master $MASTER \
+--conf spark.executor.cores=1 \
+--total-executor-cores 4 \
+--class com.intel.analytics.zoo.examples.tfnet.Predict \
```
Comment (qiyuangong) [inline zoo/src/main/scala/com/intel/analytics/zoo/examples/run-scala-example-test.sh]: Wrong class name. :)

File: zoo/src/main/scala/com/intel/analytics/zoo/examples/run-scala-example-test.sh [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+
+export SPARK_HOME=$SPARK_HOME
+export MASTER=local[4]
+export FTP_URI=$FTP_URI
+export ANALYTICS_ZOO_ROOT=$ANALYTICS_ZOO_ROOT
+export ANALYTICS_ZOO_HOME=$ANALYTICS_ZOO_ROOT/dist
+export ANALYTICS_ZOO_JAR=`find ${ANALYTICS_ZOO_HOME}/lib -type f -name "analytics-zoo*jar-with-dependencies.jar"`
+export ANALYTICS_ZOO_PYZIP=`find ${ANALYTICS_ZOO_HOME}/lib -type f -name "analytics-zoo*python-api.zip"`
+export ANALYTICS_ZOO_CONF=${ANALYTICS_ZOO_HOME}/conf/spark-analytics-zoo.conf
+export PYTHONPATH=${ANALYTICS_ZOO_PYZIP}:$PYTHONPATH
```
Comment (hkvision) [inline zoo/src/main/scala/com/intel/analytics/zoo/examples/run-scala-example-test.sh]: Since you are using script, are these variables needed?

**Comment by qiyuangong** [comment]
Please rebase with master.

File: zoo/src/main/scala/com/intel/analytics/zoo/examples/run-scala-example-test.sh [hunk 3]
```diff
@@ -0,0 +1,321 @@
...
+${ANALYTICS_ZOO_ROOT}/apps/model-inference-examples/model-inference-flink/target/model-inference-flink-0.1.0-SNAPSHOT-jar-with-dependencies.jar  \
+--modelType resnet_v1_50 --checkpointPath resnet_v1_50.ckpt  \
+--image ${ANALYTICS_ZOO_ROOT}/zoo/src/test/resources/imagenet/n04370456/ \
+--classes ${ANALYTICS_ZOO_ROOT}/zoo/src/main/resources/imagenet_classname.txt  \
+--inputShape "1,224,224,3" --ifReverseInputChannels true --meanValues "123.68,116.78,103.94" --scale 1
+
+./flink-1.7.2/bin/stop-cluster.sh
+
+echo "Finish test"
+
```
Comment (hkvision) [inline zoo/src/main/scala/com/intel/analytics/zoo/examples/run-scala-example-test.sh]: Add some timing records?

**Comment by hkvision** [comment]
Shall we maintain a record which Scala example tests are run with robot and which are newly added in this PR?

**Comment by respecteverything** [comment]
PR pass : http://10.239.47.211:18888/view/ZOO-PR/job/ZOO-PR-Scala-Simple-Integration/28/

**Comment by respecteverything** [comment]
> Shall we maintain a record which Scala example tests are run with robot and which are newly added in this PR? (probably write in our wiki?

I think we can list all examples both python and scala and mark the examples which are added in auto test?

**Comment by hkvision** [comment]
> > Shall we maintain a record which Scala example tests are run with robot and which are newly added in this PR? (probably write in our wiki?
> 
> I think we can list all examples both python and scala and mark the examples which are added in auto test?

We aim at all of them into auto test.
But scala also has tests here https://github.com/intel-analytics/analytics-zoo/blob/master/zoo/src/test/integration-test.robot and in your script. If in the future we want to trace which are added, new comers may not aware the two places.

**Pull request merged**